pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -211,10 +209,12 @@ func (cb *ClusterBuilder) buildSubsetCluster(opts buildClusterOpts, destRule *co
 
 	// Apply traffic policy for subset cluster with the destination rule traffic policy.
 	opts.mutable = subsetCluster
-	opts.istioMtlsSni = defaultSni
 
 	// If subset has a traffic policy, apply it so that it overrides the destination rule traffic policy.
 	opts.policy = MergeTrafficPolicy(opts.policy, subset.TrafficPolicy, opts.port)
+	if opts.policy != nil && opts.policy.GetTls().GetMode() == networking.ClientTLSSettings_ISTIO_MUTUAL {
+		opts.istioMtlsSni = model.BuildDNSSrvSubsetKey(model.TrafficDirectionOutbound, subset.Name, service.Hostname, opts.port.Port)
```

Comment:
I think this is not backward compatible if the server relies on this value.

Reply:
We are not using this field while building upstreamTLSContext.

Reply:
OK, i read wrong